app/models/manageiq/providers/cloud_manager/vm.rb

```diff
@@ -308,6 +307,19 @@ def self.evacuate_queue(userid, vm, options = {})
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
 
+  def resize_queue(userid, options = {})
+    options[:src_ids] = [id]
+    # TODO: @requestId - figure out what to do. old code in resize:
+    #   if params[:id] && params[:id] != 'new'
+    #     @request_id = params[:id]
+    #   end
+    VmCloudReconfigureRequest.make_request(@request_id, options, current_user)
+  end
+
+  def current_user
+    @current_user ||= User.current_user
+  end
```

Comment:
```suggestion
  def resize_queue(userid, options = {})
    options[:src_ids] = [id]
    user = options.delete(:user)
    # TODO: @requestId - figure out what to do. old code in resize:
    #   if params[:id] && params[:id] != 'new'
    #     @request_id = params[:id]
    #   end
    VmCloudReconfigureRequest.make_request(@request_id, options, user)
  end
```

Reply:
@agrare 
Just wanted to clarify this.
Are you suggesting that I send it in through the front end options payload?

I guess another idea would be to instead of sending in the userid to the resize_queue, we can just send in the current_user

in manager-api
```def resize_resource(type, id, data)
      api_resource(type, id, "Resizing Resource", :supports => :resize) do |vm|
        raise BadRequestError, "Must specify new resize value/s" if data["resizeValues"].blank?

        {:task_id => vm.resize_queue(User.current_user, data)}
      end
    rescue => err
      action_result(false, err.to_s)
    end```

or maybe if we want to keep the queue methods consistent by always sending in a userid, we can just do a direct look up from User in this resize_queue method.

Reply:
@alizapeikes the typical prototype for a queue method looks like `def update_cloud_tenant_queue(userid, options = {})` and the API calls it with `klass.create_cloud_tenant_queue(User.current_userid, ems, data)}` so I think it'd be best to stick to that pattern

Reply:
@agrare 
Yes, that makes sense. So when you say we should send it in through options, you mean through the front end form code?

Reply:
You can just pass it in as a parameter to this method so keep the:
```
def resize_queue(userid, options = {})
```

And call it from the API/UI like:
```
vm.resize_queue(User.current_userid, data)
```

Reply:
@agrare 
I originally started by doing what you just suggested:
```vm.resize_queue(User.current_userid, data)```
However, the resize request goes through the vm_cloud_reconfigure_request class: `VmCloudReconfigureRequest.make_request(@request_id, options, user)` 
which requires current_user to be passed in, not the userid?

Reply:
oh, ok perfect.